### ISSUE TITLE
Omicron crashes in a strange way

### ISSUE THREAD

**Issue by jan-warchol** [mcFrax/omicron-dnd#10]
**Describe the bug**

![image](https://user-images.githubusercontent.com/1368508/152675038-e9936789-dcf4-4b60-8c6a-1404482096ed.png)

**To Reproduce**
Not sure how. This appeared when I tried to grab a task and drop it fast into another.

**Comment by mcFrax** [comment]
I think it should be fixed by 8917c6a3ad5, which added a condition to `setPlaceholderAndNoMoveZone`.

On the first screenshot, there are 3 errors in console - the first one is #8, the third on is `setPlaceholderAndNoMoveZone`, and these are fixed.

I'm still concerned about the middle one, as it appears that we have `newIndex > toEl.children.length`. This really shouldn't happen, even with some loose placeholders dangling at the end - maybe unless there are some external changes to the DOM, but I don't suspect that. However, it may be a side effect of one of the above failures - perhaps toEl was updated and newIndex wasn't, or the other way around - in either case, it might happen that we end up with this error. I'll leave that one for now.

**Comment by jan-warchol** [comment]
Ok. I'll get back to you if I see it again. Thanks!

**Comment by mcFrax** [comment]
I reproduced the issue with placeholder not being removed. As I suspected, the issue was an extension of what I fixed in #11.

It turns out my assumption about pointer capture was wrong - only for touch the pointer is automatically captured, but not for mouse. That meant that we could get pointerleave and pointerenter events during preDrag - specifically, when starting the drag at the edge of an item and dragging quickly into some other container (e.g. elements-list on the card one-up in Organizer), we could "enter" another container before starting the drag (i.e. still in the minimalMoveMouse range). That then messed up the drag initialization logic (assumption that toEl is null and there is no placeholder).

Explicitly capturing the pointer helps (tested on desktop, will test on mobile too, to make sure it doesn't break in some stupid way). I'll make the fix tomorrow.
